project3/participant/participant.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file Multicast Participant Class
+ */
+#ifndef PROJECT3_PARTICIPANT_H
+#define PROJECT3_PARTICIPANT_H
+
+#include <cstdint>
+
+#include "../../common/thread_pool/task.h"
+#include "../../common/thread_pool/thread_pool.h"
+#include "input_parser/input_parser.h"
+#include "pub_sub_messages.pb.h"
+#include "tasks/console_task.h"
+#include "tasks/multicast_receiver_task.h"
+
+namespace participant {
+class Participant {
+ public:
+  /**
+   * @param config_loc the location of the config file
+   */
+  explicit Participant(const std::string& config_loc);
+
+  /**
+   * Start the participant
+   */
+  void Start();
+
+ private:
+  /**
+   * @param config_loc the config file location
+   */
+  void LoadConfig(const std::string& config_loc);
+
+  /**
+   * @param msg the message to send to the coordinator
+   * @return successfully sent
+   */
+  bool OneOff(const google::protobuf::Message& msg);
+
+  /// Pool and tasks
+  thread_pool::ThreadPool pool_;
+  std::shared_ptr<participant_tasks::ConsoleTask> console_task_;
+  std::shared_ptr<participant_tasks::MulticastReceiver> multicast_receiver_;
```

Comment:
Note: this is one example of a design compromise I made to satisfy project 3 requirements. If this has to be expanded on for a later project, I expect to refactor this accordingly.